src/rust/ide/lib/analytics/src/remote_log.rs

```diff
@@ -0,0 +1,21 @@
+//! Provides an API to send data to our remote logging service. Requires the remote logging
+//! to be already set up on the JS side. That means, there needs to exist a `window.enso.remote_log`
+//! method that takes a string and does the actual logging.
+
+use js_sys;
+use wasm_bindgen::JsValue;
+
+/// Send the provided public event to our logging service.
+pub fn remote_log(message:crate::data::Public) {
+    let js_new  = js_sys::Function::new_with_args("msg", "window.enso.remote_log(msg)");
+    let context = JsValue::NULL;
```

Comment:
creating functions is VERY PRICY this way. Remote logging will happen frequently. This way runs JS parser and compiler on every call and also breaks some cache, this is very bad and could completely kill our performance (!). Instead, use standard wasm_bindgen API to connect to JS functions (foreign calls).